clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -583,11 +582,24 @@ void CodeGenFunction::EmitKernelMetadata(const FunctionDecl *FD,
     && !FD->hasAttr<SYCLDeviceAttr>())
     return;
 
-  // TODO Module identifier is not reliable for this purpose since two modules
-  // can have the same ID, needs improvement
-  if (getLangOpts().SYCLIsDevice)
+  
+  if (getLangOpts().SYCLIsDevice) {
+    // TODO Module identifier is not reliable for this purpose since two modules
+    // can have the same ID, needs improvement
     Fn->addFnAttr("sycl-module-id", Fn->getParent()->getModuleIdentifier());
-
+    int SYCLDeviceCompileOptLevel = -1;
+    switch (CGM.getCodeGenOpts().OptimizationLevel) {
+    default:
+      llvm_unreachable("Invalid optimization level!");
+    case 0:
+    case 1:
+    case 2:
+    case 3:
+      SYCLDeviceCompileOptLevel = CGM.getCodeGenOpts().OptimizationLevel;
+    }
```

Comment:
```suggestion
    int SYCLDeviceCompileOptLevel = CGM.getCodeGenOpts().OptimizationLevel;
    if (SYCLDeviceCompileOptLevel < 0) ...;
```
Why not (something like) this instead?  I am trying to understand the benefit of checking the 0,1,2,3 values in the switch.